tests/test_index.py

```diff
@@ -149,6 +149,38 @@ def test_similarity_errors(self, indexed_img_tbl: pxt.Table, small_img_tbl: pxt.
             _ = t.order_by(t.split.similarity(sample_img)).limit(1).collect()
         assert "was created without the 'image_embed' parameter" in str(exc_info.value).lower()
 
+    def test_add_index_after_drop(self, small_img_tbl: pxt.Table, reload_tester: ReloadTester) -> None:
+        """ Test the an index with the same name can be added after the previous one is dropped """
+        skip_test_if_not_installed('transformers')
+        t = small_img_tbl
+        sample_img = t.select(t.img).head(1)[0, 'img']
+        t.add_embedding_index('img', idx_name='clip_idx', image_embed=clip_img_embed, string_embed=clip_text_embed)
+        orig_res = t.select(t.img.localpath).order_by(t.img.similarity(sample_img, idx='clip_idx'), asc=False).limit(3).collect()
+        t.revert()
+        # creating an index with the same name again after a revert should be successful
+        t.add_embedding_index('img', idx_name='clip_idx', image_embed=clip_img_embed, string_embed=clip_text_embed)
+        res = t.select(t.img.localpath).order_by(t.img.similarity(sample_img, idx='clip_idx'), asc=False).limit(3).collect()
+        assert_resultset_eq(orig_res, res, True)
+        t.revert()
+        # should be true even after reloading from persistence
+        reload_tester.run_reload_test()
```

Comment:
This isn't doing anything other than calling reload_catalog().

Reply:
Yes, that was my intent. That even after reloading, the state is clean and the another index by same name can be created. I am replacing reload_catalog() with reload_test.run_reload_test() in new tests (even when no queries are to be verified). Let me know if we prefer to keep using reload_catalog() in such cases.

FWIW, t.revert() did not have any issue. The bug was only in t.drop_index(). But I retained revert() also in the final testing to keep the code coverage.
<!-- Reviewable comment -OEBWG5w7Wh4h96BQ_P0:btmdsqj -->
<!-- Sent from Reviewable.io -->

Reply:
@mkornacker Is there an explicit preference for calling `reload_catalog()` directly if there are no queries to test after reload? (In this case there are no queries to test since `revert()` was called before the reload.)
<!-- Reviewable comment -OEQYWHH3H4axh7KamUE:b-jjuyfi -->
<!-- Sent from Reviewable.io -->

Reply:
If you don't run any queries through reload_tester, there is nothing to test.

In that case, calling reload_catalog() directly is clearer.

Reply:
Done. 
<!-- Reviewable comment -OESRBbgDERuebd9SgOq:bs7wumx -->
<!-- Sent from Reviewable.io -->